cmd/server/http/jwt.go

```diff
@@ -0,0 +1,151 @@
+package http
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	httpinternal "github.com/lunarway/release-manager/internal/http"
+	"github.com/lunarway/release-manager/internal/log"
+	"github.com/pkg/errors"
+)
+
+const keyNotFoundMsg = "failed to find key with key ID"
+
+type JwkCache interface {
+	Get(ctx context.Context, url string) (jwk.Set, error)
+	Refresh(ctx context.Context, url string) (jwk.Set, error)
+}
+
+type Verifier struct {
+	jwksLocation string
+	issuer       string
+	audience     string
+
+	jwkCache JwkCache
+}
+
+func NewVerifier(jwksLocation string, jwkFetchTimeout time.Duration, issuer string, audience string) (*Verifier, error) {
+	ctx := context.Background()
+
+	cache := jwk.NewCache(ctx)
+	err := cache.Register(jwksLocation, jwk.WithMinRefreshInterval(24*time.Hour))
+	if err != nil {
+		return nil, err
+	}
+	_, err = cache.Refresh(ctx, jwksLocation)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Verifier{
+		jwksLocation: jwksLocation,
+		jwkCache:     cache,
+		issuer:       issuer,
+		audience:     audience,
+	}, nil
+}
+
+func ParseBearerToken(token string) (string, error) {
+	jwt := strings.TrimPrefix(token, "Bearer")
+
+	tokenParts := strings.Split(jwt, ".")
+
+	if len(tokenParts) != 3 {
+		return "", errors.New("invalid token format")
+	}
+
+	return strings.TrimSpace(jwt), nil
+}
+
+const AUTH_USER_KEY = "AUTH_USER_KEY"
```

Comment:
A best practice is to have a complex type to avoid anyone outside the package to access the subject.

```suggestion
type AUTH_USER_KEY struct{}
```